frontend/schemes/DynamicOpenIDConnectScheme.js

```diff
@@ -47,10 +63,11 @@ export default class DynamicOpenIDConnectScheme extends OpenIDConnectScheme {
     }
 
     async updateAccessToken() {
-      if (!this.idToken.sync()) {
+      if (this.isValidMealieToken()) {
         return
       }
-      if (this.isValidMealieToken()) {
+      if (!this.idToken.status().valid()) {
+        this.idToken.reset();
         return
```

Comment:
I don't think the second if is needed here. Token is either valid or invalid, so I think you can just remove the indent for lines 69 and 70.